### PR TITLE
Enable to set callback_url via environment variables

### PR DESCRIPTION
**Pull request by mats16**
### Motivation and context

The callback url is hardcoded, so I cannot  run CVAT on Cloud platform.
This patch enable to replace these parameters via environment variables.

#5526

### How has this been tested?

Run on AWS Fargate with Google auth.

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Review by nmanovic — DISMISSED** [review]
LGTM

**Comment by nmanovic** [comment]
@mats16 , thanks for your contribution!

**Comment by Marishka17** [comment]
@nmanovic, @mats16, I guess we should use `CVAT_HOST` environment variable because `auth/login-with-social-app` , `auth/<provider>/login/callback/` are part of our server routes and should be immutable.

**Comment by mats16** [comment]
I think it is better to use CVAT_HOST too. But CVAT_HOST does not have schema.

**Comment by mats16** [comment]
Can I add the environmental variable such as ‘CVAT_EXTERNAL_URL’ and use it instead of  ‘CVAT_HOST’?

**Comment by nmanovic** [comment]
> @nmanovic @Marishka17 I think it is better to use `CVAT_HOST` too. But `CVAT_HOST` does not have schema.
> 
> Can I add the environmental variable such as `CVAT_EXTERNAL_URL` and use it instead of `CVAT_HOST`?

@mats16 , there is one more opportunity. You can import our settings file into your my_settings.py and redefine what even you want. Will it work for you?

**Comment by mats16** [comment]
@nmanovic 
> there is one more opportunity. You can import our settings file into your my_settings.py and redefine what even you want. Will it work for you?

I feel it is not good option... Because CVAT project provide container images officially, it is better to configure via environment variables. I want use official container images.

If you need more time to discuss about `CVAT_HOST`, it is better to release once as`GOOGLE_CALLBACK_URL`.
`CALLBACK_URL: localhost:8080` obviously does not work.

**Comment by Marishka17** [comment]
@mats16, Why is the approach of using `CVAT_HOST` + another environment variable (e.g `CVAT_SCHEME` or `USE_HTTPS`) worse than adding `GITHUB_CALLBACK_URL`,  `GOOGLE_CALLBACK_URL`,  `SOCIAL_APP_LOGIN_REDIRECT_URL`?
We can add the scheme env variable in docker-compose.https.yml and use it to define the scheme in our settings (default http).

**Comment by mats16** [comment]
I don't think adding other environment variables is so bad. This is just my personal opinion...

- When publishing a web application over the internet, a schema (such as https) is required and it is almost never necessary to use only hostname.
- If you want to host CAVT under a specific url path (such as https://example.com/cvat/), you cannot configure via CVAT_SCHEMA and CVAT_HOST environment variables. You need another variables.

How about bellow?

```py
CVAT_HOST = os.getenv('CVAT_HOST', 'localhost')
CVAT_BASE_URL = os.getenv('CVAT_EXTERNAL_URL', 'http://{}:8080'.format(CVAT_HOST))
```

**Comment by Marishka17** [comment]
@mats16, Hi, sorry for the long response. Okay, it looks good to me.

File: cvat/settings/base.py [hunk 1]
```diff
@@ -636,17 +639,17 @@ class CVAT_QUEUES(Enum):
...
+    GITHUB_CALLBACK_URL = f'{CVAT_BASE_URL}/api/auth/github/login/callback/'
+    GOOGLE_CALLBACK_URL = f'{CVAT_BASE_URL}/api/auth/google/login/callback/'
+    
     SOCIAL_AUTH_GOOGLE_CLIENT_ID = os.getenv('SOCIAL_AUTH_GOOGLE_CLIENT_ID')
     SOCIAL_AUTH_GOOGLE_CLIENT_SECRET = os.getenv('SOCIAL_AUTH_GOOGLE_CLIENT_SECRET')
 
     SOCIAL_AUTH_GITHUB_CLIENT_ID = os.getenv('SOCIAL_AUTH_GITHUB_CLIENT_ID')
     SOCIAL_AUTH_GITHUB_CLIENT_SECRET = os.getenv('SOCIAL_AUTH_GITHUB_CLIENT_SECRET')
-
+
```
Comment (nmanovic) [inline cvat/settings/base.py:652]:
```suggestion
```

**Comment by mats16** [comment]
Sorry, I fixed it.

**Review by nmanovic — APPROVED** [review]

**Comment by nmanovic** [comment]
@mats16 , thanks for the contribution!

**Pull request merged**